java/flight/flight-jdbc-driver/src/main/java/org/apache/arrow/driver/jdbc/accessor/impl/text/ArrowFlightJdbcVarCharVectorAccessor.java

```diff
@@ -100,9 +100,15 @@ public String getString() {
   }
 
   @Override
-  public boolean getBoolean() {
+  public boolean getBoolean() throws SQLException {
     String value = getString();
-    return value != null && !value.isEmpty() && !value.equals("false") && !value.equals("0");
+    if (value == null || value.equalsIgnoreCase("false") || value.equals("0")) {
+      return false;
+    } else if (value.equalsIgnoreCase("true") || value.equals("1")) {
+      return true;
+    } else {
+      throw new SQLException("Is not possible to convert this value for boolean: " + value);
```

Comment:
Small change on the message -   "It is not possible..."

Reply:
fixed